{{cookiecutter.profile_name}}/OSLayer.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+import subprocess
+from typing import Tuple
+import random
+import string
+
+
+class OSLayer:
+    """
+    This class provides an abstract layer to communicating with the OS.
+    Its main purpose is to enable OS operations mocking, so we don't actually need to make file operations or create
+    processes.
+    """
+    @staticmethod
+    def mkdir(directory: Path):
+        directory.mkdir(parents=True, exist_ok=True)
+
+    @staticmethod
+    def remove_file(file: Path):
+        if file.exists():
+            file.unlink()
+
+    @staticmethod
+    def run_process_and_get_output_and_error_stream(cmd: str) -> Tuple[str, str]:
+        completed_process = subprocess.run(
+            cmd, check=True, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        return completed_process.stdout.decode().strip(), completed_process.stderr.decode().strip()
+
+    @staticmethod
+    def print(string: str):
+        print(string)
+
+    @staticmethod
+    def get_random_alphanumerical_string(n : int = 64) -> str:
```

Comment:
If this function is just for testing purposes, maybe indicate as being private (i.e. `_get_random_alpha....`)

Reply:
It is not for testing purposes only. The name of the log files are the jobid + this random string: https://github.com/leoisl/snakemake-lsf/blob/5155b82cf3d3b2b87cbcd6de31ad28a4fe6766e0/%7B%7Bcookiecutter.profile_name%7D%7D/lsf_submit.py#L119-L125

The purpose of this random string is if a pipeline is executed several times (due to some jobs failing, and then several resumes), different jobs can get the same id in different executions of the pipeline.
For example, let's say that in execution 1 of the pipeline, job pandora_compare with id=10 fails, while job pandora_index with id=5 succeeds. Some changes are made, and in the execution 2, pandora_compare gets assigned id=5. In this case, if bjobs fail, we could inadvertently read that the job with id=5 (i.e. pandora_compare) succeeded, while in fact it was the previous pandora_index.  This random 64 alphanumerical string is to avoid these conflicts with jobs with same ids in different reexecutions of the pipeline. It is not a perfect solution though, someone could be very unlucky of having the same job id with the same random 64-length alphanumerical string, so this has to be improved.